### PR TITLE
Rewrite core feature specs

### PR DESCRIPTION
**Pull request by tijmenb**
The feature specs describe the functionality of this application. Over the last year this app has gotten more responsibilities, growing the these tests.

It has some problems:

- Some are really large (~ 350 lines). It's hard to quickly scan which functionality is being tested.
- They are hard to follow. Some of the tests have Cucumber-like comments (`When I make a change to the mainstream browse page`), but other don't.
- Lots of duplication in boilerplate code (create a record, navigate to this page)
- Duplicated tests. Many specs test a slightly different code branch, or test the same functionality twice.

These problems make the test suite hard to work with. A small change can break many tests at once, and it is hard to figure out why they are failing. It is then challenging to figure out if the tests or the code should change, resulting in shaky tests.

This PR rewrites the core feature specs for this app. It splits up the functionality into features and scenarios, using methods to implement the different steps. By combining examples (use the same scenario for creating and editing a record) we reduce of code necessary.

This is a technique also used in content-tagger and explained in this blog post: 

https://about.futurelearn.com/blog/how-we-write-readable-feature-tests-with-rspec/

### How to review

- Do the scenarios make sense for a new developer coming on this project? Suggestions on the naming of the steps very welcome.
- Do the scenarios cover all of the known functionality?

### CODE REVIEW

File: spec/support/common_feature_steps.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  def given_I_am_not_a_GDS_editor
+    stub_user.permissions = ['signin']
+  end
+
+  def and_I_submit_the_form
+    find('input[type=submit]').click
+  end
+
+  def then_I_see_a_message_that_I_dont_have_access
```
Comment (mobaig) [inline spec/support/common_feature_steps.rb]: Slight mismatch between method name and implementation - perhaps `then_i_am_informed_i_dont_have_access`?

File: spec/features/archiving_tags_spec.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+  scenario "User redirects to invalid basepath" do
+    given_there_is_a_published_topic
+    and_I_visit_the_topic
+    and_I_go_to_the_archive_page
+    when_I_submit_an_invalid_base_path_as_redirect
+    then_I_see_that_the_URL_isnt_valid
+  end
+
+  def and_external_services_are_stubbed
```
Comment (mobaig) [inline spec/features/archiving_tags_spec.rb]: Intentional override of the method in CommonFeatureSteps?

**Comment by mobaig** [comment]
Big :+1: for intent. Couple of observations, but LGTM.

File: spec/features/access_control_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.feature "Access control" do
+  include CommonFeatureSteps
+  include NavigationSteps
+
+  scenario "Non-GDS editor navigates browse pages" do
```
Comment (stephen-richards) [inline spec/features/access_control_spec.rb]: "Non-GDS editor cannot navigate browse pages" might be a better name

**Comment by stephen-richards** [comment]
Vast improvement and a model for future feature specs

File: spec/features/access_control_spec.rb (hunk 3 above)
Comment (tijmenb) [inline spec/features/access_control_spec.rb]: Thanks, updated.

**Comment by tijmenb** [comment]
Updated branch with feedback and quashed the commits.
